### PR TITLE
bpo-38722: Runpy use io.open_code()

### PR DESCRIPTION
**Pull request by jsnklln**
https://bugs.python.org/issue38722

Automerge-Triggered-By: @taleinat

### CODE REVIEW

**Review by brandtbucher — APPROVED** [review]
Thanks @jsnklln!

**Comment by brandtbucher** [comment]
@zooba

**Review by taleinat — APPROVED** [review]

File: Misc/NEWS.d/next/Security/2019-11-18-16-17-56.bpo-38722.x3mECW.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+:mod:`runpy` now uses :meth:`io.open_code` which will trigger auditing events.
```
Comment (taleinat) [inline Misc/NEWS.d/next/Security/2019-11-18-16-17-56.bpo-38722.x3mECW.rst]: `open()` and `io.open()` also trigger auditing events. And as far as I can tell, this won't change anything WRT auditing; it only allows setting an "open code hook" using [PyFile_SetOpenCodeHook](https://docs.python.org/3.8/c-api/file.html#c.PyFile_SetOpenCodeHook). So this text needs to be revised.

**Review by taleinat — CHANGES_REQUESTED** [review]

File: Misc/NEWS.d/next/Security/2019-11-18-16-17-56.bpo-38722.x3mECW.rst (hunk 1 above)
Comment (jsnklln) [reply Misc/NEWS.d/next/Security/2019-11-18-16-17-56.bpo-38722.x3mECW.rst]: I've updated with a better description of what the change really means/does.

**Review by zooba — APPROVED** [review]
Looks good to me, but let's wait on the backport discussion before merging.

*Edit:* Missed the automerge tag - oh well :) We can still apply the backport tag if we decide to add to 3.8.1.

**Comment by brandtbucher** [comment]
Thanks @jsnklln!

**Comment by taleinat** [comment]
Thanks, @zooba! Since this is apparently a security-related issue, I've backported it.

**Pull request merged**